src/nlsolve/anderson.jl

```diff
@@ -131,26 +136,33 @@ function (S::NLAnderson{true})(integrator)
     if mass_matrix == I
       @. z₊ = dt*k
     else
+      ztmp = b
       @. z₊ = dt*k + z
       mul!(ztmp, mass_matrix, z)
       @. z₊ -= ztmp
     end
-    gs[:,1] .= vec(z₊)
+    gs[1] .= vec(z₊)
 
     mk = min(S.n, iter-1)
-    residuals[:,1:mk] .= (gs[:,2:mk+1] .- zs[:,2:mk+1]) .- (gs[:,1] - zs[:,1])
-    alphas[1:mk] .= residuals[:,1:mk] \ (zs[:,1] .- gs[:,1])
+    ztmp = vec(b)
+    @. ztmp = zs[1] - gs[1]
+    for i in 2:mk+1
+      @. residuals[:,i-1] = gs[i] - zs[i] + ztmp
+    end
+    alphas[1:mk] .= @view(residuals[:,1:mk]) \ ztmp
```

Comment:
should use `ldiv!` here, and maybe not the default factorization. Is this where the singularity comes from?